### PR TITLE
Add no-auth option

### PR DESCRIPTION
**Pull request by AsafMah**
Solves #398

### CODE REVIEW

File: data/src/main/java/com/microsoft/azure/kusto/data/http/HttpRequestBuilder.java [hunk 1]
```diff
@@ -107,13 +107,25 @@ public HttpRequestBuilder withURL(String url) throws DataClientException {
...
     }
 
-    public HttpRequest build() {
+    public HttpRequest build() throws DataClientException {
+        // If has authorization header, ensure it is not sent over insecure channel
+        boolean hasAuth = request.getHeaders().stream().anyMatch(h -> h.getName().equalsIgnoreCase(HttpHeaderName.AUTHORIZATION.toString()));
+        if (hasAuth) {
+            URL url = request.getUrl();
+            boolean isHttp = url.getProtocol().equalsIgnoreCase("http");
+            boolean isLocalhost = url.getHost().toLowerCase().startsWith(CloudInfo.LOCALHOST);
```
Comment (yogilad) [inline data/src/main/java/com/microsoft/azure/kusto/data/http/HttpRequestBuilder.java:116]: this is a bit risky.
localhostmyass.com is a valid target 😅
You'd want to regex match the host against `^localhost(\:[0-9]+)?$`

File: data/src/main/java/com/microsoft/azure/kusto/data/http/HttpRequestBuilder.java [hunk 2]
```diff
@@ -107,13 +107,25 @@ public HttpRequestBuilder withURL(String url) throws DataClientException {
...
+
+            if (isHttp && !isLocalhost) {
+                throw new DataClientException(url.toString(), "Cannot forward security token to a remote service over insecure " +
```
Comment (yogilad) [inline data/src/main/java/com/microsoft/azure/kusto/data/http/HttpRequestBuilder.java:119]: insecure means someone who is not confident in himself.
Unsecure, not secure, unenceypted, etc.

**Review by yogilad — COMMENTED** [review]
Looks good.
Left some comments.
You can commit after fixing

File: data/src/main/java/com/microsoft/azure/kusto/data/http/HttpRequestBuilder.java (hunk 1 above)
Comment (AsafMah) [reply data/src/main/java/com/microsoft/azure/kusto/data/http/HttpRequestBuilder.java:116]: good point (though this is the logic we already have).

I think the host method returns it without the port, so I'll just check equality

File: data/src/main/java/com/microsoft/azure/kusto/data/http/HttpRequestBuilder.java [hunk 3]
```diff
@@ -107,13 +107,24 @@ public HttpRequestBuilder withURL(String url) throws DataClientException {
...
-    public HttpRequest build() {
+    public HttpRequest build() throws DataClientException {
+        // If has authorization header, ensure it is not sent over insecure channel
+        boolean hasAuth = request.getHeaders().stream().anyMatch(h -> h.getName().equalsIgnoreCase(HttpHeaderName.AUTHORIZATION.toString()));
+        if (hasAuth) {
+            URL url = request.getUrl();
+            boolean isHttp = url.getProtocol().equalsIgnoreCase("http");
+            boolean isLocalhost = url.getHost().equalsIgnoreCase(CloudInfo.LOCALHOST);
+
+            if (isHttp && !isLocalhost) {
```
Comment (ohadbitt) [inline data/src/main/java/com/microsoft/azure/kusto/data/http/HttpRequestBuilder.java:118]: maybe log the case of isHttp + isLocalHost 
so users would be aware

**Pull request merged**